File: market-realtime/src/main/java/com/finsight/marketrealtime/model/Subscription.java
package com.finsight.marketrealtime.model;

import com.finsight.marketrealtime.enums.SubscriptionEnum;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Subscription {

    @Id
    private Long subscriptionId;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private String errorMessage;
    @Enumerated(EnumType.STRING)
    public SubscriptionEnum status;


    @ManyToOne
    private UserEntity user;

    @ManyToOne
    private SubscriptionPlanEntity subscriptionPlan;
}

File: market-rest/src/models/Subscription.ts
import type { UserEntity } from "./UserEntity";
import type { SubscriptionPlanEntity } from "./SubscriptionPlanEntity";

export enum SubscriptionEnum {
    ACTIVE = "ACTIVE",
    EXPIRED = "EXPIRED",
    CANCELED = "CANCELED",
}

export interface Subscription {
    subscriptionId: string;
    startDate: string;
    endDate: string;
    status: SubscriptionEnum;
    user: UserEntity;
    subscriptionPlan: SubscriptionPlanEntity;
}

